packages/svelte/src/compiler/phases/3-transform/client/visitors/template.js

```diff
@@ -2445,60 +2449,80 @@ export const template_visitors = {
 	},
 	SnippetBlock(node, context) {
 		// TODO hoist where possible
+		/** @type {import('estree').Pattern[]} */
 		const args = [b.id('$$anchor')];
 
 		/** @type {import('estree').BlockStatement} */
 		let body;
 
-		if (node.context) {
-			const id = node.context.type === 'Identifier' ? node.context : b.id('$$context');
-			args.push(id);
+		/** @type {import('estree').Statement[]} */
+		const declarations = [];
 
-			/** @type {import('estree').Statement[]} */
-			const declarations = [];
+		node.context.elements.forEach((argument, i) => {
+			if (!argument) return;
 
-			// some of this is duplicated with EachBlock — TODO dedupe?
-			if (node.context.type === 'Identifier') {
+			if (argument.type === 'Identifier') {
+				args.push(argument);
 				const binding = /** @type {import('#compiler').Binding} */ (
-					context.state.scope.get(id.name)
+					context.state.scope.get(argument.name)
 				);
-				binding.expression = b.call(id);
-			} else {
-				const paths = extract_paths(node.context);
+				binding.expression = b.call(argument);
+				return;
+			}
 
-				for (const path of paths) {
-					const name = /** @type {import('estree').Identifier} */ (path.node).name;
-					const binding = /** @type {import('#compiler').Binding} */ (
-						context.state.scope.get(name)
-					);
+			let arg_alias = `$$arg${i}`;
+
+			if (argument.type === 'RestElement') {
+				args.push(b.rest(b.id(arg_alias)));
+
+				if (argument.argument.type === 'Identifier') {
 					declarations.push(
-						b.let(
-							path.node,
-							b.thunk(
-								/** @type {import('estree').Expression} */ (
-									context.visit(path.expression?.(b.call('$$context')))
+						b.let(argument.argument.name, b.call('$.proxy_rest_array', b.id(arg_alias)))
```

Comment:
Why proxy the rest array? For convenience because we don't have to change reads/writes later on? What would the alternative look like? We have handling for this in other places already so it shouldn't be impossible to solve differently, if we think using a proxy has drawbacks.

Reply:
I tried to explain in the PR description -- I couldn't think of another way to do it. If you have ideas I'm all ears!